src/core/vectortile/qgsvectortilewriter.cpp

```diff
@@ -319,3 +362,18 @@ QByteArray QgsVectorTileWriter::writeSingleTile( QgsTileXYZ tileID, QgsFeedback
   return encoder.encode();
 }
 
+
+QgsTileMatrix QgsVectorTileWriter::tileMatrixBuilder( const int &zoomLevel )
+{
+  const QgsCoordinateReferenceSystem crs = mZ0TileMatrix.crs();
+  if ( mZ0TileMatrix.crs().authid() == mDefaultCrs )
+  {
+    return QgsTileMatrix::fromWebMercator( zoomLevel );
+  }
+  const QgsRectangle r = mZ0TileMatrix.extent();
+  return QgsTileMatrix::fromCustomDef( zoomLevel, crs,
+                                       QgsPointXY( r.xMinimum(), r.yMaximum() ),
+                                       r.width() / mZ0TileMatrix.matrixWidth(),
+                                       mZ0TileMatrix.matrixWidth(),
+                                       mZ0TileMatrix.matrixHeight() );
+}
```

Comment:
It would be quite useful to have this method directly in `QgsTileMatrix` - it may be useful elsewhere to get derived tile matrix for a lower zoom level. Also, there should be not need for a special case for web mercator, it should be just a matter of multiplying/dividing the root tile's values